internal/core/graphics/brush.rs

```diff
@@ -258,16 +270,260 @@ impl RadialGradientBrush {
+                } else {
+                    last.color
+                };
+                stops.insert(0, GradientStop { position: 0.0, color: color_at_0 });
+            }
+        }
+
+        let has_stop_at_1 = stops.iter().any(|s| (s.position - 1.0).abs() < f32::EPSILON);
+        if !has_stop_at_1 {
+            if let Some(first) = stops.first() {
+                stops.push(GradientStop { position: 1.0, color: first.color });
+            }
+        }
+
+        // Update the internal storage
+        self.0 = SharedVector::with_capacity(stops.len() + 1);
+        self.0.push(GradientStop { color: Default::default(), position: from_angle });
+        self.0.extend(stops.into_iter());
+    }
+
+    /// Returns the starting angle (rotation) of the conic gradient in degrees.
+    pub fn angle(&self) -> f32 {
+        self.0[0].position
     }
 
     /// Returns the color stops of the conic gradient.
+    /// The stops are already rotated according to the `from_angle` specified in `new()`.
     pub fn stops(&self) -> impl Iterator<Item = &GradientStop> {
-        self.0.iter()
+        // skip the first fake stop that just contains the angle
+        self.0.iter().skip(1)
+    }
+
+    /// Helper: Linearly interpolate between two colors using premultiplied alpha.
+    ///
+    /// This is used for interpolating gradient boundary colors in CSS-style gradients.
+    /// We cannot use Color::mix() here because it implements Sass color mixing algorithm,
+    /// which is different from CSS gradient color interpolation.
+    ///
+    /// CSS gradients interpolate in premultiplied RGBA space:
+    /// https://www.w3.org/TR/css-color-4/#interpolation-alpha
+    fn interpolate_color(c1: &Color, c2: &Color, factor: f32) -> Color {
+        let argb1 = c1.to_argb_u8();
+        let argb2 = c2.to_argb_u8();
+
+        // Convert to premultiplied alpha
+        let a1 = argb1.alpha as f32 / 255.0;
+        let a2 = argb2.alpha as f32 / 255.0;
+        let r1 = argb1.red as f32 * a1;
+        let g1 = argb1.green as f32 * a1;
+        let b1 = argb1.blue as f32 * a1;
+        let r2 = argb2.red as f32 * a2;
+        let g2 = argb2.green as f32 * a2;
+        let b2 = argb2.blue as f32 * a2;
+
+        // Interpolate in premultiplied space
+        let alpha = (1.0 - factor) * a1 + factor * a2;
+        let red = (1.0 - factor) * r1 + factor * r2;
+        let green = (1.0 - factor) * g1 + factor * g2;
+        let blue = (1.0 - factor) * b1 + factor * b2;
+
+        // Convert back from premultiplied alpha
+        if alpha > 0.0 {
+            Color::from_argb_u8(
+                (alpha * 255.0) as u8,
+                (red / alpha).min(255.0) as u8,
+                (green / alpha).min(255.0) as u8,
+                (blue / alpha).min(255.0) as u8,
+            )
+        } else {
+            Color::from_argb_u8(0, 0, 0, 0)
+        }
+    }
+}
+
+/// C FFI function to normalize the gradient stops to be within [0, 1] range
+#[no_mangle]
```

Comment:
```suggestion
#[unsafe(no_mangle)]
```
(in preparation for the 2024 edition)